### PR TITLE
Jeremy/labswap

### PR DESCRIPTION
**Pull request by Unit404**
Labswap:
    Contains the labswap components and all of the copied+changed scheduling components used for it.
    Labswap page displays similar to the schedule view but compresses the courses. 
    Allows user to click on a course to select a section for trading.
TODO:
    add context/state for tracking user info and selections
    integrate API
    add options for selecting specific people to trade with

### CODE REVIEW

File: src/Components/LabSwap/CourseBlock.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        IconComponent={()=>null}
+      >
+        <MenuItem value="" ><em>None</em></MenuItem>
+        {course_instance.section_numbers.map((v, i) => <MenuItem value={v.toString()} key={i}>{v}</MenuItem>)}
+      </Select>
+    </FormControl>
+  )
+  // FIXME: Questionable Key (using the index) here?
+}
+
```
Comment (benbeauchamp7) [inline src/Components/LabSwap/CourseBlock.tsx:99]: 👀

File: src/Components/LabSwap/Selection.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import React, { FC } from 'react'
```
Comment (benbeauchamp7) [inline src/Components/LabSwap/Selection.tsx:1]: Filename should start with the main component it's related to. ie. SwapSelection

File: src/Components/Scheduling/SchedulingFilter.tsx [hunk 3]
```diff
@@ -15,7 +19,7 @@ const base = {
 }
 
 export const SchedulingFilter: FC<Props> = ({ filter, setFilter }) => {
-  let filter_subjects = ['121', '221', '222', '312', '313', '314', '315'];
+  let filter_subjects = ['110', '111', '120', '121', '206', '221', '222', '312', '313', '314', '315'];
   const showAll = 'Show All'
   let filter_elements = [];
```
Comment (benbeauchamp7) [inline src/Components/Scheduling/SchedulingFilter.tsx:24]: Hardcoded filters can be fixed later

File: src/Components/Scheduling/SchedulingWindow.tsx [hunk 4]
```diff
@@ -9,7 +9,11 @@ interface Props {
...
 export const SchedulingWindow: FC<Props> = () => {
   const [filter, setFilter] = useState({
+    110: true,
+    111: true,
+    120: true,
     121: true,
+    206: true,
     221: true,
     222: true,
     312: true,
```
Comment (benbeauchamp7) [inline src/Components/Scheduling/SchedulingWindow.tsx:19]: Hardcoded filters can be fixed later

**Review by benbeauchamp7 — APPROVED** [review]
Lots of stuff to go next sprint, but looks good enough to at least demo tomorrow

**Review by dantebarbieri — APPROVED** [review]
LGTM &ndash; But the select *None* option didn't work as I expected.

**Pull request merged**